### DISCUSSION TITLE
hi pls help

### DISCUSSION THREAD

**Discussion by kpopdev** [eritislami/evobot#353] [Q&A]
i keep getting this error (node:8328) UnhandledPromiseRejectionWarning: Error [SHARDING_READY_TIMEOUT]: Shard 1's Client took too long to become ready. 
someone said if i add onready event on index,js or add triggerReady it will fix it but idk how.
can someone tell me pls.bot keep dying.

**Comment by BlazeIsClone** [comment]
can u post some of ur code? 

> i keep getting this error (node:8328) UnhandledPromiseRejectionWarning: Error [SHARDING_READY_TIMEOUT]: Shard 1's Client took too long to become ready.
> someone said if i add onready event on index,js or add triggerReady it will fix it but idk how.
> can someone tell me pls.bot keep dying.

can u post some of ur code?

**Reply by kpopdev** [reply]
it fix it was my shard.js settings
but sure here : const config = require('./config.json')

const { ShardingManager } = require('discord.js');
const shard = new ShardingManager('./index.js', {
  token: config.TOKEN,
  autoSpawn: true
});

shard.spawn();

shard.on('launch', shard => console.log(`[SHARD] Shard ${shard.id}/${shard.totalShards}`));

the problem was shard timeout and i set it to infinty and it fix.
